linera-core/src/client/mod.rs

```diff
@@ -362,17 +372,20 @@ impl<Env: Environment> Client<Env> {
 
     async fn download_blobs(
         &self,
-        remote_node: &RemoteNode<impl ValidatorNode>,
-        blob_ids: impl IntoIterator<Item = BlobId>,
+        remote_nodes: &[RemoteNode<Env::ValidatorNode>],
+        blob_ids: &[BlobId],
     ) -> Result<(), ChainClientError> {
         self.local_node
             .store_blobs(
-                &futures::stream::iter(blob_ids.into_iter().map(|blob_id| async move {
-                    remote_node.try_download_blob(blob_id).await.unwrap()
-                }))
-                .buffer_unordered(self.options.max_joined_tasks)
-                .collect::<Vec<_>>()
-                .await,
+                &self
+                    .validator_manager
+                    .download_blobs(remote_nodes, blob_ids, self.options.blob_download_timeout)
+                    .await?
+                    .ok_or_else(|| {
+                        ChainClientError::RemoteNodeError(NodeError::BlobsNotFound(
+                            blob_ids.to_vec(),
+                        ))
+                    })?,
```

Comment:
I'd put that in a local variable before we call `store_blobs`.

Reply:
OK, but that was already here – I just moved it around.